google-cloud-clients/google-cloud-firestore/src/main/java/com/google/cloud/firestore/FirestoreOptions.java

```diff
@@ -208,6 +216,32 @@ public Builder setTimestampsInSnapshotsEnabled(boolean value) {
     @Override
     @Nonnull
     public FirestoreOptions build() {
+      if (System.getProperty(FIRESTORE_EMULATOR_SYSTEM_PROPERTY) != null) {
```

Comment:
I wasted the last 10 minutes on this, and I couldn't come to a conclusion. If it is possible to change environment variables after the fact (it looks like it might be on windows) then checking this twice might be racy. But it's probably not.